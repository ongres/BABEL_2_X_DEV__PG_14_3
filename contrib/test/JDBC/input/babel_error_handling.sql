CREATE TABLE t1 ( a int primary key );
GO

INSERT INTO t1 VALUES (1);
GO

--
-- Basic error handling behaviors
--

CREATE PROCEDURE proc_insert1 @a int AS
BEGIN
INSERT INTO t1 VALUES (@a);
IF @@error = 2627
    PRINT 'DUPLICATE KEY DETECTED';
IF @@error = 0
    PRINT 'INSERTED @a'

INSERT INTO t1 VALUES (@a+1);

IF @@error = 2627
    PRINT 'DUPLICATE KEY DETECTED @a+1';
IF @@error = 0
    PRINT 'INSERTED @a+1';
END
GO

--
-- entire transaction will be aborted
-- no rows will be inserted
--

EXEC proc_insert1 1
GO

SELECT * FROM t1 ORDER BY a;
GO

-- sub-transaction is aborted and outer transaction continues
EXEC proc_insert1 1
GO

SELECT * FROM t1 ORDER BY a;
GO

-- clean up
DROP TABLE t1;
GO
DROP PROCEDURE proc_insert1;
GO
