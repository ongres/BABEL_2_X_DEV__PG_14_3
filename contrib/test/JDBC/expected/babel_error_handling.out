CREATE TABLE t1 ( a int primary key );
GO

INSERT INTO t1 VALUES (1);
GO
~~ROW COUNT: 1~~





--
-- Basic error handling behaviors
--
CREATE PROCEDURE proc_insert1 @a int AS
BEGIN
INSERT INTO t1 VALUES (@a);
IF @@error = 2627
    PRINT 'DUPLICATE KEY DETECTED';
IF @@error = 0
    PRINT 'INSERTED @a'
INSERT INTO t1 VALUES (@a+1);
IF @@error = 2627
    PRINT 'DUPLICATE KEY DETECTED @a+1';
IF @@error = 0
    PRINT 'INSERTED @a+1';
END
GO


--
-- entire transaction will be aborted
-- no rows will be inserted
--
EXEC proc_insert1 1
GO
~~ERROR (Code: 2627)~~

~~ERROR (Message: duplicate key value violates unique constraint "t1_pkey")~~


SELECT * FROM t1 ORDER BY a;
GO
~~START~~
int
1
2
~~END~~


-- sub-transaction is aborted and outer transaction continues
EXEC proc_insert1 1
GO
~~ERROR (Code: 2627)~~

~~ERROR (Message: duplicate key value violates unique constraint "t1_pkey")~~


SELECT * FROM t1 ORDER BY a;
GO
~~START~~
int
1
2
~~END~~


-- clean up
DROP TABLE t1;
GO
DROP PROCEDURE proc_insert1;
GO
